File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int smallest = Integer.MAX_VALUE;
        int profit = 0;

        for(int price : prices){
            profit = Math.max(profit,price-smallest);
            if(smallest > price){
                smallest = price;
            }
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let lowest = prices[0];
    let maxProfit = 0;

    prices.forEach(function(price){
        maxProfit = Math.max(maxProfit,price-lowest);
        if(price < lowest){
            lowest = price;
        }
    })
    return maxProfit;
};